devito/symbolics/extended_sympy.py

```diff
@@ -262,8 +263,8 @@ def __new__(cls, params):
         for p in as_tuple(params):
             if isinstance(p, str):
                 args.append(Symbol(p))
-            elif isinstance(p, int):
-                args.append(p)
+            elif isinstance(p, (int, np.integer)):
```

Comment:
you're looking for
```
from devito.tools import is_integer
```

Reply:
What happens if p is already an Integer?

The `is_integer` also tests for this case and this is the reason why I didn't use it.

Reply:
In [11]: from sympy import Integer                                                                                                                                                                                 

In [12]: Integer(Integer(2))                                                                                                                                                                                       
Out[12]: 2

Reply:
It still seems a little awkward to create an Integer if it is already an Integer. 

The Integer case will still be added by the else condition here.

But I'm OK with change it to use is_integer if there is no problem in the tests.

Reply:
so the point is in theory we could at some point have other objects that behave "like" integers, but are not any of sympy/numpy/int . And if we use `is_integer`, the idea is that we capture them all. So it's "safer", put it that way

Reply:
Fixed.